MapboxNavigation/InstructionPresenter.swift

```diff
@@ -43,16 +43,18 @@ class InstructionPresenter {
         
         guard !stringFits else { return attributedPairs.attributedStrings }
         
-        let indexedComponents = attributedPairs.components.enumerated().map { IndexedVisualInstructionComponent(component: $1, index: $0) }
-        let filtered = indexedComponents.filter { $0.component.abbreviation != nil }
-        let sorted = filtered.sorted { $0.component.abbreviationPriority < $1.component.abbreviationPriority }
-        for component in sorted {
-            let isFirst = component.index == 0
+        let filteredComponents = attributedPairs.components.compactMap { $0 as? VisualInstructionComponent}
+        let visuals = filteredComponents.enumerated().map { IndexedVisualInstructionComponent(component: $1, index: $0) }
```

Comment:
Can there be anything except `VisualInstructionComponent` in attributed pairs? If that's the case, the instruction indices can be shuffled and end up in the wrong order with this change. Indexing should be done before anything else.

Reply:
@frederoni  Nope. There are only `VisualIntructionComponent`s in `IndexedVisualInstructionComponent`. See: https://github.com/mapbox/mapbox-navigation-ios/pull/1464/files#diff-af48d70309ee195e34cc5e6cc6b27caaR300

Reply:
What I meant was that this row: 46
```swift
let filteredComponents = attributedPairs.components.compactMap { $0 as? VisualInstructionComponent}
```
filters out components before we index them.
Further down, where we abbreviate the string, it could end up in the wrong order, no?

Reply:
Gotchu.

Reply:
@frederoni so, wouldn't the sorting by abbreviation priority address this concern?
https://github.com/mapbox/mapbox-navigation-ios/pull/1464/files/b8f447cc296ce7cc9866794c62a5183d8990f543#diff-af48d70309ee195e34cc5e6cc6b27caaR49

Or, perhaps what would you suggest will be the ideal approach in this context?